AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm

```diff
@@ -139,7 +148,28 @@
     initializedNoteStates = false;
     aePlayingNotes.polyphony = S1_MAX_POLYPHONY;
 
-    AKPolyphonicNode.tuningTable.middleCFrequency = getSynthParameter(frequencyA4) * exp2((60.f - 69.f)/12.f);
-
     // initializeNoteStates() must be called AFTER init returns, BEFORE process, turnOnKey, and turnOffKey
 }
+
+
+// private tuningTable lookup
+double S1DSPKernel::tuningTableNoteToHz(int noteNumber) {
+
+    const int nn = clamp(noteNumber, 0, 127);
+    return getTuningTable(nn);
+}
+
+// S1TuningTable protocol
+void S1DSPKernel::setTuningTable(float frequency, int index) {
+    tuningTable[index].store(frequency);
```

Comment:
Is it worth clamping the index here as well like above?

Reply:
Yes